app/assets/javascripts/discourse/app/controllers/fullscreen-code.js

```diff
@@ -0,0 +1,24 @@
+import Controller from "@ember/controller";
+import { schedule } from "@ember/runloop";
+import ModalFunctionality from "discourse/mixins/modal-functionality";
+import highlightSyntax from "discourse/lib/highlight-syntax";
+import CodeblockButtons from "discourse/lib/codeblock-buttons";
+
+export default Controller.extend(ModalFunctionality, {
+  onShow() {
+    schedule("afterRender", () => {
```

Comment:
You can now use `@afterRender` decorator in controllers 😃

Reply:
Nice, will change

Reply:
@CvX this doesn't seem to work, the other times we have used this has been on components. This code in the decorator function gets hit, but the `afterRender` scheduled call never gets hit. Maybe because the controller doesn't have afterRender, do you think it's worth adding a whole component for this?

https://github.com/discourse/discourse/blob/6ebdeca53c5e7cdb818947682563658991ec6634/app/assets/javascripts/discourse-common/addon/utils/decorators.js#L21-L30

Reply:
I get a 404 on that link.

The `@afterRender` change landed in core very recently, have you tried with the latest `main`?

Reply:
Oh that's weird I got 404. Yes I have latest `main` and it still doesn't work, this is what the modal controller code looks like:

```javascript
import Controller from "@ember/controller";
import { afterRender } from "discourse-common/utils/decorators";
import { schedule } from "@ember/runloop";
import ModalFunctionality from "discourse/mixins/modal-functionality";
import highlightSyntax from "discourse/lib/highlight-syntax";
import CodeblockButtons from "discourse/lib/codeblock-buttons";

export default Controller.extend(ModalFunctionality, {
  onClose() {
    this.codeBlockButtons.cleanup();
  },

  @afterRender
  _applyCodeblockButtons() {
    const modalElement = document.querySelector(".modal-body");

    highlightSyntax(modalElement, this.siteSettings, this.session);

    this.codeBlockButtons = new CodeblockButtons({
      showFullscreen: false,
      showCopy: true,
    });
    this.codeBlockButtons.attachToGeneric(modalElement);
  },
});
```

Reply:
Ohhhh okay I was so confused :see_no_evil: Thanks, will fix up and merge